src/System.Collections/tests/Generic/Stack/Stack.Generic.Tests.cs

```diff
@@ -227,7 +215,28 @@ public void Stack_Generic_TrimExcess_AfterClearingAndAddingAllElementsBack(int c
                 Assert.Equal(count, stack.Count);
             }
         }
-
-        #endregion
+        
+        // Contains
+        
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Stack_Generic_Contains_BasicFunctionality(int count)
+        {
+            Stack<T> stack = GenericStackFactory(count);
+            List<T> list = stack.ToList();
+            
+            // Stack should contain all items that result from enumeration
+            Assert.All(list, item => Assert.True(stack.Contains(item)));
+            
+            stack.Clear();
+            
+            // Stack should not contain any items after being cleared
+            Assert.All(list, item => Assert.False(stack.Contains(item)));
+            
+            foreach (T item in list) stack.Push(item);
+            
+            // Stack should contain whatever items are pushed to it
+            Assert.All(list, item => Assert.True(stack.Contains(item)));
+        }
```

Comment:
@stephentoub Well for one, it's capturing the stack into a list before asserting that it contains each element, so for example if a logic error is introduced into Stack's `GetEnumerator` method it won't affect the `Contains` tests (which currently [foreach over the collection itself](https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Collections/IGenericSharedAPI.Tests.cs#L279-L280)). Also to my knowledge the behavior of `Contains` after `Clear` and `Push` are not tested in the base class.